hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/handlers/MarkerHandler.java

```diff
@@ -0,0 +1,482 @@
+          Set<String> allMarkers = allMarkersMap.computeIfAbsent(markerDirPath, k -> new HashSet<>());
+          boolean exists = allMarkers.contains(markerName);
+          if (!exists) {
+            allMarkers.add(markerName);
+            StringBuilder stringBuilder = fileMarkersMap.computeIfAbsent(markerDirPath, k -> new HashMap<>())
+                .computeIfAbsent(markerFileIndex, k -> new StringBuilder(16384));
+            stringBuilder.append(markerName);
+            stringBuilder.append('\n');
+            updatedMarkerDirPaths.add(markerDirPath);
+          }
+          future.setResult(!exists);
+          futuresToRemove.add(future);
+        }
+        createMarkerFutures.removeAll(futuresToRemove);
+      }
+      LOG.info("Flush to MARKERS file .. ");
+      flushMarkersToFile(updatedMarkerDirPaths, markerFileIndex);
+      markerFilesUseStatus.set(markerFileIndex, false);
+      LOG.info("Resolve request futures .. ");
+      for (CreateMarkerCompletableFuture future : futuresToRemove) {
+        try {
+          future.complete(jsonifyResult(future.getContext(), future.getResult(), metricsRegistry, OBJECT_MAPPER, LOG));
+        } catch (JsonProcessingException e) {
+          throw new HoodieException("Failed to JSON encode the value", e);
+        }
+      }
+      LOG.info("Finish batch processing of create marker requests in " + (System.currentTimeMillis() - startTimeMs) + " ms");
+    }
+
+    private void flushMarkersToFile(Set<String> updatedMarkerDirPaths, int markerFileIndex) {
+      long flushStartTimeMs = System.currentTimeMillis();
+      for (String markerDirPath : updatedMarkerDirPaths) {
+        LOG.info("Write to " + markerDirPath);
+        long startTimeMs = System.currentTimeMillis();
+        Path markersFilePath = new Path(markerDirPath, MARKERS_FILENAME_PREFIX + markerFileIndex);
+        Path dirPath = markersFilePath.getParent();
+        try {
+          if (!fileSystem.exists(dirPath)) {
+            fileSystem.mkdirs(dirPath);
+          }
+        } catch (IOException e) {
+          throw new HoodieIOException("Failed to make dir " + dirPath, e);
+        }
+        FSDataOutputStream fsDataOutputStream = null;
+        BufferedWriter bufferedWriter = null;
+        try {
+          LOG.info("Create " + markersFilePath.toString());
+
+          fsDataOutputStream = fileSystem.create(markersFilePath);
+          bufferedWriter = new BufferedWriter(new OutputStreamWriter(fsDataOutputStream, StandardCharsets.UTF_8));
+          bufferedWriter.write(fileMarkersMap.get(markerDirPath).get(markerFileIndex).toString());
+        } catch (IOException e) {
+          throw new HoodieIOException("Failed to overwrite marker file " + markersFilePath, e);
+        } finally {
+          closeQuietly(bufferedWriter);
+          closeQuietly(fsDataOutputStream);
+        }
+        LOG.info(markersFilePath.toString() + " written in " + (System.currentTimeMillis() - startTimeMs) + " ms");
```

Comment:
Can you revisit all logging and fix info, debug levels as appropriate.

Reply:
Fixed.